### ISSUE TITLE
Rules documentation

### ISSUE THREAD

**Issue by dschnare** [dudleycarr/ratelimit.js#42]
I love you guys for implementing this piece of software, but where can I find more details on the rules themselves? I tried reading the source code and your two part article (linked to from ratelimit.io), but I'm still struggling to understand them.

Say for example:

```js
const rules = [
  { interval: 1, limit: 5 },
  { interval: 3600, limit: 1000, precision: 10000 }
]
```

Is this adding two separate rate limiting counting strategies with the following rules:

*Assumptions:*
- `interval` is in seconds
- `limit` is number of successful requests
- `precision` is number of tokens/buckets to break up the interval into

**rules[0]**

For every one second a consumer is permitted 5 requests (count is reset at the top of the second)

**rules[1]**

For every one hour, there will be 10000 buckets where a consumer will be permitted 1000 
requests per bucket (count for each bucket will be reset one hour after the bucket's timeslice)

So basically, by doing `new RateLimiter(client, rules)` we'd effectively be getting two rate limiters since we have two rules for counting.

**Comment by dschnare** [comment]
After re-reading the Sliding Window in [article 2](http://www.dr-josiah.com/2014/11/introduction-to-rate-limiting-with_26.html) and the lua source code again I have another take on the interpretation of `rules[1]`. I believe `precision: 10000` is actually an error in the example because in the lua code any value larger than the `interval` will result in the `precision` being set to the `interval`.

https://github.com/dudleycarr/ratelimit.js/blob/master/lua/check_limit.lua#L8-L9

Here's an example to illustrate my new understanding:

**Assumptions:**
- `interval` is in seconds
- `limit` is number of successful requests *per bucket*
- `precision` *is the duration of a bucket in seconds that will divide the interval* (must be <= interval)

**Given a rule like:** `{ interval: 3600, limit: 1000, precision: 10 }`
**Then the rule means:** For every one hour there will be 360 buckets (one every 10 seconds), where a consumer can make 1000 requests per bucket (i.e. 10 seconds).

**Update:** In the example `precision` is actually set to `100` not `1000`. I misread the example. However, this would just mean that the example `rule[1]` would mean *For every 100 seconds over a one hour period a consumer can make 1000 requests*.

**Comment by dschnare** [comment]
After writing my own unit tests I now realize what is going on with the rules.

**Given**
```
{ interval: 3600, limit: 1000, precision: 120 }
```

**Then**
A consumer can make 1000 requests per hour, but there are actually only 30 buckets (one at each 120 second mark) that can be used within a one hour window. I'm a little confused as to the use of the limit in sliding window, but I think I understand this enough to use this little module now.
